tests/conftest.py

```diff
@@ -909,6 +911,40 @@ def create_mock_mqtt(*args, **kwargs) -> MqttMockHAClient:
         yield _setup_mqtt_entry
 
 
+@pytest.fixture
+def yaml_config() -> ConfigType | None:
+    """Fixture to override configuration.yaml using mock_yaml_config."""
+    return None
+
+
+@pytest.fixture
+def mock_yaml_config(
+    hass: HomeAssistant, tmp_path: Path, yaml_config: ConfigType
+) -> Generator[Path, None, None]:
+    """Fixture to mock the configuration.yaml file content.
+
+    Parameterize configuration.yaml using the `yaml_config` fixture.
+    """
+    new_yaml_config_file = tmp_path / "configuration.yaml"
+    new_yaml_config_file.write_text(yaml.dump(yaml_config or {}))
+
+    with patch.object(hass_config, "YAML_CONFIG_FILE", new_yaml_config_file):
+        yield new_yaml_config_file
```

Comment:
So you need to return the `Path`?
Should this use `patch_yaml_files` instead?
```suggestion
    files = {YAML_CONFIG_FILE: yaml_config}
    with patch_yaml_files(files):
        yield
```

Reply:
Does patch_yaml_files translate the `dict` `yaml_config` into yaml?

Reply:
Added a test to prove the yaml override code working